### PR TITLE
Fixes bug 1486509 - Add search and status filters to Translate.Next.

### PR DESCRIPTION
**Pull request by adngdb**
This feature is not complete on purpose. I'm going to file follow-up bugs for missing features: 

- [x] select several filters
- [x] extra filters (rejected, unchanged, untranslated )
- [x] tags
- [x] author
- [x] time range

See the commits for the detail of each change. I've tried to be better with those, and cut the PR in meaningful sets of changes.

### CODE REVIEW

**Comment by mathjazz** [comment]
I haven't look at the code yet, so this is just an early observation:
* Search results don't get [highlighted](https://pontoon.mozilla.org/sl/amo/all-resources/?search=Firefox&string=154956).
* If no matches are found, the [No results](https://pontoon.mozilla.org/sl/amo/all-resources/?search=qwerty) message doesn't show up.
* When you hover over the Missing string in the string list or Missing filter in the filter menu, the Missing icon "disappears" (prolly uses the same color as the hover background).

**Comment by adngdb** [comment]
>     * Search results don't get [highlighted](https://pontoon.mozilla.org/sl/amo/all-resources/?search=Firefox&string=154956).

I've filed [bug 1511357](https://bugzilla.mozilla.org/show_bug.cgi?id=1511357) for this. The other two issues have been resolved. :-)

File: frontend/src/modules/search/components/SearchBox.css [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    box-sizing: border-box;
+    color: #fff;
+    float: left;
+    font-size: 16px;
+    font-weight: 300;
+    line-height: 23px;
+    padding: 10px 35px 10px 43px;
+    width: 100%;
+
+    /* Remove extra padding in Chrome search input */
```
Comment (mathjazz) [inline frontend/src/modules/search/components/SearchBox.css]: We have other webkit specific styles in `translate.css`. For example, in prod we remove the clear/cancel button in the input field that Chrome shows.

We should also use this opportunity to make the placeholder text in Chrome use the same color as in Firefox.

File: frontend/src/modules/search/components/SearchBox.css [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.search-box .unreviewed .status:before {
+    color: #4FC4F6;
+    content: "";
+}
+
+.search-box .missing:hover .status:before {
+    color: #4D5967;
+}
```
[the remark this answers is not in the document]
Comment (adngdb) [reply frontend/src/modules/search/components/SearchBox.css]: We should, and we should use CSS Variables for that. However I think it's fine to move on with imperfect CSS for now. I've been thinking about styles for some time now, I'll get to a PR were I refactor those not too far in the future.

**Comment by mathjazz** [comment]
OK, thanks for the comments! I guess we should file bugs to avoid losing track of that?

**Pull request merged**